### PR TITLE
Doc/getting started blocks storage

### LINKED ISSUE #2206: Document getting started with blocks storage
We should add a new guide "Getting Started - Blocks storage" setting up Cortex single binary configured with GCP/S3.

### PR DESCRIPTION
**Pull request by AndreZiviani**
**What this PR does**:
Create a `getting started` documentation for blocks storage.

**Which issue(s) this PR fixes**:
Fixes #2206

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/configuration/single-process-config-blocks.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    # Use an in memory ring store, so we don't need to launch a Consul.
+    ring:
+      kvstore:
+        store: inmemory
+      replication_factor: 1
+
+storage:
+  engine: tsdb
+
+  tsdb:
```
Comment (thorfour) [inline docs/configuration/single-process-config-blocks.yaml]: This looks incorrectly indented. Tsdb is a top level config object.

File: docs/configuration/single-process-config-blocks.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    max_recv_msg_size: 104857600
+    max_send_msg_size: 104857600
+    use_gzip_compression: true
+
+ingester:
+  lifecycler:
+    # The address to advertise for this ingester.  Will be autodiscovered by
+    # looking up address on eth0 or en0; can be specified if this fails.
+    # address: 127.0.0.1
+
```
Comment (thorfour) [inline docs/configuration/single-process-config-blocks.yaml]: I think we recommend running block storage as stateful, so we typically turn off transfers
`max_transfer_retries: 0`

File: docs/configuration/single-process-config-blocks.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  engine: tsdb
+
+  tsdb:
+    backend: s3
+    dir: /tmp/cortex/tsdb
+    bucket_store:
+      sync_dir: /tmp/cortex/tsdb-sync
+    s3:
+      bucket_name: cortex
+      endpoint: s3.dualstack.us-east-1.amazonaws.com
```
Comment (thorfour) [inline docs/configuration/single-process-config-blocks.yaml]: Might be a good idea to add a compactor section?

File: docs/configuration/single-process-config-blocks.yaml (hunk 1 above)
Comment (AndreZiviani) [reply docs/configuration/single-process-config-blocks.yaml]: my bad

File: docs/configuration/single-process-config-blocks.yaml (hunk 2 above)
Comment (AndreZiviani) [reply docs/configuration/single-process-config-blocks.yaml]: added `max_transfer_retries: 0`

File: docs/configuration/single-process-config-blocks.yaml (hunk 3 above)
Comment (AndreZiviani) [reply docs/configuration/single-process-config-blocks.yaml]: done

File: docs/getting-started/getting-started-blocks.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+This document will focus on single-process Cortex with the experimental blocks storage. See [the architecture doc](../architecture.md) for more information about the microservices and [blocks operation](../operations/blocks-storage.md) for more information about the blocks storage.
+
+Separately from single process vs microservices decision, Cortex can be configured to use local storage or cloud storage (S3, GCS and Azure).
+This document will focus on using S3.
+Cortex can also make use of external memcacheds and redis for caching but this feature is not available (yet) using block storage.
+
+## Single instance, single process
+
+For simplicity & to get started, we'll run it as a [single process](../configuration/single-process-config-blocks.yaml) with no dependencies:
```
Comment (gouthamve) [inline docs/getting-started/getting-started-blocks.md]: `simplicity & to` --> `simplicity and to`

**Comment by gouthamve** [comment]
Thanks a ton for this work! This looks good!

**Pull request merged**